lib/mongo/auth/scram/conversation.rb

```diff
@@ -429,9 +429,15 @@ def xor(first, second)
           first.bytes.zip(second.bytes).map{ |(a,b)| (a ^ b).chr }.join('')
         end
 
+        def secure_compare(a, b)
+          check = a.bytesize ^ b.bytesize
+          a.bytes.zip(b.bytes){ |x, y| check |= x ^ y.to_i }
```

Comment:
What does zip do if the string lengths aren't equal? MIght be better to take the hash from the server and compare it to itself in that case.